src/plugins/profiling/README.md

```diff
@@ -62,3 +62,51 @@ You should expect some of these commands to take several minutes to finish.
 ## Development
 
 See the [kibana contributing guide](https://github.com/elastic/kibana/blob/main/CONTRIBUTING.md) for instructions setting up your development environment.
+
+### Update dev branch with changes from upstream
+
+As of this writing, we use the branch `8.2+profiler` for the development of Kibana profiling plugin under `src/plugins/profiling/`.
+
+Make sure the remote `upstream` exists. If it doesn't, create it with `git remote add upstream https://github.com/elastic/kibana`.
+
+To update to the latest `8.2` from upstream:
+- get latest changes from all remotes with `git fetch --all --prune`
+- execute `git checkout 8.2+profiler` and `git pull` to make sure you have the latest changes from `optimyze/kibana`
+- get the latest changes from upstream with `git fetch upstream`
+- update with `git rebase upstream/8.2` and push with `git push --force-with-lease`
+
+### Upgrade dev branch to next Kibana version
+
+From time to time we want to upgrade the Kibana version for development to the latest release version.
+
+Make sure the remote `upstream` exists. If it doesn't, create it with `git remote add upstream https://github.com/elastic/kibana`.
+
+Example on how upgrade from 8.2 to 8.3:
+- get latest changes from all remotes with `git fetch --all --prune`
+- create and checkout the new development branch with `git checkout -b 8.3+profiler upstream/8.3`
+- cherry-pick all our changes from `8.2+profiler` with e.g. `git cherry-pick 10fdea5a7156ca0d576142e78ab0d17dc7420b1d^..8.2+profiler`, where that commit hash refers to our first commit (created by Stephanie at 07 Dec "add initial profiling search").
+- fix the merge conflicts that are likely popping up. Normally, these are in `package.json` and `yarn.lock`.
+- when done, push the new branch to our development repository with `git push --set-upstream origin 8.3+profiler`
+
+### Build and push the ES cloud docker image
```

Comment:
We should add here the pre-requisite of logging into the registry, linking the infra doc (it's mentioned in the eng@ thread with subject "Environment for MVP"

Reply:
:+1: Updated